tools/monitoring/landsat_gap_report.py

```
@@ -1 +1,310 @@
+            f"*{landsat.upper()} GAP REPORT*\n "
+            f"Environment: {environment}\n "
+            f"Missing Scenes: {len(missing_scenes)}\n"
+            f"Orphan Scenes: {len(orphaned_scenes)}\n"
+            f"Missing Scenes report Saved: {str(URL(landsat_status_report_path) / output_filename)}\n"
+            f"Orphan Scenes report Saved: {str(URL(landsat_status_report_path) / orphan_output_filename)}\n"
+        )
+
+        if notification_url is not None and (
+            len(missing_scenes) > 0 or len(orphaned_scenes) > 0
+        ):
+            send_slack_notification(notification_url, f"{landsat} Gap Report", message)
+
+        log.info(message)
+        log.info(
+            f"File {file_name} processed and sent in {time_process(start=start_timer)}"
+        )
+
+        if (
+            len(missing_scenes) > 200 or len(orphaned_scenes) > 200
+        ) and not update_stac:
+            sys.exit(1)
+
+    except Exception as error:
+        logging.error(error)
+        # print traceback but does not stop execution
+        traceback.print_exc()
+        raise error
+
+
+@click.argument(
+    "bucket_name",
+    type=str,
+    nargs=1,
+    required=True,
+    default="Bucket where the gap report is",
+)
+@click.argument(
+    "satellite",
+    type=str,
+    nargs=1,
+    required=True,
+    default="satellite to be compared, supported ones (landsat_8, landsat_7, landsat_5)",
+)
+@update_stac
+@slack_url
+@click.command("landsat-gap-report")
+def cli(
+    bucket_name: str, satellite: str, update_stac: bool = False, slack_url: str = None
+):
+    """
+    Publish missing scenes
+    """
+    try:
+        generate_buckets_diff(
+            bucket_name=bucket_name,
+            landsat=satellite,
+            file_name=FILES.get(satellite, None),
+            update_stac=update_stac,
+            notification_url=slack_url,
+        )
+    except Exception as error:
+        traceback.print_exc()
+        raise error
```

Catching an exception and raising an exception is not useful. It's redundant. Just let the exception be raised.

Please don't do this.